docs/proposals/20230715-improve-yurthub-cache.md

```diff
@@ -0,0 +1,547 @@
+| filter        | s cachemanager.StorageWrapper<br/>nodeKey, err := s.KeyFunc(storage.KeyBuildInfo{<br/>&emsp;&emsp;Component: "kubelet",<br/>&emsp;&emsp;Name:      ssf.nodeName,<br/>&emsp;&emsp;Resources: "nodes",<br/>&emsp;&emsp;Group:     "",<br/>&emsp;&emsp;Version:   "v1",<br/>})<br/>obj, err := s.Get(nodeKey) | get kubelet node                                             |
+| proxy         | 1. localProxy uses QueryCache to query data when processing get/list/update requests offline, first check the cache, if it doesn't exist, then get it through Storage<br/>2. poolProxy/loadbalanceProxy tries to query data from QueryCache if there is an error in the cloud response to a request when performing reverse proxy. |                                                              |
+
+-   YurtHub yaml file because there are two modes of cloud side, so there are two kinds of files, only a small inconsistency in the input parameter, the rest are the same, it is easy to not see the problem without a detailed comparison, and it is easy to make mistakes when upgrading.
+-   In the original schema, the storage of list data generated by different invocations of path or selector on the same component and the same resource is anomalously conflicting, as an example:
+
+```shell
+request1: http://{ip/port}/api/v1/pods?fieldSelector=spec.nodeName=foo
+request2: http://{ip/port}/api/v1/default/pods?fieldSelector=spec.nodeName=foo
+request3: http://{ip:port}/api/v1/default/pods?labelSelector=foo2=bar2
+
+The three requests above:
+1 and 2 belong to the same resource pod under the same selector list query request, but the difference is that the path is different, that is, the query namespace is different
+2 and 3 belong to the same resource pod under the same path list query request, but the difference is that the selector is different, that is, the query conditions are different
+
+If the same component, such as kubelet, has multiple modules, each module needs to access ApiServer to perform list queries for pod resources, and then store the returned results to local disk. However, the list conditions of each module are different, the original yurthub practice is to store only the first list data locally, and retain the relevant list conditions in memory, and then come back to the list request, if the conditions are not consistent, will not be stored!
+```
+
+### Goals
+
+-   Sorting out the original architecture logic, identifying problems and proposing solutions
+-   Optimized architectural design, weakened YurtHub cloud-side two patterns code design
+
+
+## Proposal
+
+### Problem Solutions
+
+#### Architecture Improvements
+
+<img src = "../img/yurthub-cache-improve/YurtHub-arch-improve.png" width="800">
```

Comment:
1. For Edge mode, Yurthub should proxy the list/get requests to kube-apiserver for clients(such as kubelet/kube-proxy, etc.) when cloud-edge network is normal. on the other hand, if cloud-edge network is offline, Yurthub should  use local cache(include HF resource caching) to handle list/get requests from clients(such as kubelet/kube-proxy, etc.) 

2. For cloud mode, Yurthub should proxy the list/get requests to kube-apiserver for clients(such as kubelet/kube-proxy, etc.) all the time.